framework/web/Request.php

```diff
@@ -197,6 +271,40 @@ public function resolve()
     }
 
     /**
+     * Filters headers according to the [[trustedHosts]].
+     * @param HeaderCollection $headerCollection
+     * @since 2.0.13
+     */
+    protected function filterHeaders(HeaderCollection $headerCollection)
+    {
+        // do not trust any of the [[secureHeaders]] by default
+        $trustedHeaders = [];
+
+        // check if the client is a trusted host
+        if (!empty($this->trustedHosts)) {
+            $host = $this->getRemoteHost();
+            $ip = $this->getRemoteIP();
+            foreach ($this->trustedHosts as $hostRegex => $headers) {
+                if (!is_array($headers)) {
+                    $hostRegex = $headers;
+                    $headers = $this->secureHeaders;
+                }
+                if (preg_match($hostRegex, $host) || preg_match($hostRegex, $ip)) {
```

Comment:
Discussion about different matching method has been discussed in https://github.com/yiisoft/yii2/pull/13780#discussion_r106126378 above already. I think its worth discussing that again.
Lets have a new issue for that.